app/app/schemas/users/user.py

```diff
@@ -13,6 +13,25 @@ class UserBase(BaseModel):
     is_admin: bool = False
     school: Optional[int] = None
 
+    @validator('type')
+    def valid_type(cls, user_type: Optional[str]) -> Optional[str]:
+        if user_type and user_type not in ("superuser", "student", "professor", "admin"):
+            raise ValueError("Invalid user type!")
+        return user_type
+
+    @validator('password')
+    def valid_password(cls, password: Optional[str]) -> Optional[str]:
+        if password and len(password) < 8:
```

Comment:
Since UserUpdate can techincally have password set as `None`, would it be wiser to move the entire check under an `if password` to make sure if password is `None` then none of the checks are carries out?

Reply:
Ah yeah, missed that.